README.md

```diff
@@ -1,8 +1,29 @@
 # hal
-Easily create and manage Kubernetes applications using Dekorate and the Halkyon operator, made with ❤️ by the Snowdrop team.
+
+## Table of Contents
+- [Overview](#overview)
+- [Key features](#key-features)
+- [Building hal](#building-hal)
+- [Downloading a snapshot](#downloading-a-snapshot)
+- [Deploying a component using `hal`](#deploying-a-component-using-hal)
+  * [1 Scaffold the Spring Boot applications](#1-scaffold-the-spring-boot-applications)
+  * [2 Deploy the Component](#2-deploy-the-component)
+  * [3 Connect to the rest services](#3-connect-to-the-rest-services)
+- [Demonstration](#demonstration)
+- [Additional documentation](additional-documentation)
+
+## Overview
+Hal is a CLI tool for developers to simplify the deployment of applications such as Spring Boot on OpenShift and Kubernetes using Dekorate and Halkyon Component Operator. Made with ❤️ by the Snowdrop team.
 
 [![CircleCI](https://circleci.com/gh/halkyonio/hal.svg?style=svg)](https://circleci.com/gh/halkyonio/hal)
 
+## Key features
+`hal` is capable of doing the following tasks
+- Scaffold Spring Boot applications
+- Deploy Microservices applications as Components
+- Switch the component from the `dev` to the `build` mode
```

Comment:
No -> Should be: Switch the DeploymentMode of the Component from `dev` to `build`